.github/workflows/ci-workflow.yml

```diff
@@ -21,6 +21,10 @@ jobs:
       run: make -f Bootstrap.mak ${{ matrix.osname }} CONFIG=${{ matrix.config }}
     - name: Test
       run: bin/${{ matrix.config }}/premake5 test
+    - uses: actions/upload-artifact@v2
+      with:
+        name: ${{ matrix.osname }}-${{ matrix.config }}
```

Comment:
Since macOS will be transitioning to ARM64 soon, it's probably worth including the platform here, adding the values into the `include` sections of `matrix`.

Reply:
Should I add platform for Linux as well? If not, it would either require an if statement, to check if the OS is MacOS or separate the job (linux-and-macos) into two.

Also wouldn't macosx look better in the filename instead of osx?

Reply:
Yeah, I think adding platform Linux is a good idea - I imagine if ARM64 desktops and laptops become the norm, people will want ARM64 builds for Linux, and possibly Windows too.

Yeah, `macosx` would look better, it's also the name we use for the release assets - I must have just copied the Travis-CI name, oops!

Reply:
I had to separate the jobs due to `matrix.include` not supporting arrays (therefore we can't spawn more jobs for x64 and ARM64).